### PR TITLE
feat: vc token api

### PR DESCRIPTION
**Pull request by skynet2**
feat: vc token api

### CODE REVIEW

**Comment by skynet2** [comment]
@fqutishat @aholovko 
image
there is such problem, but this is not related to my changes, its related to metrics from Baha, not sure why it shows impact in my pr

File: docs/v1/openapi.yaml [hunk 1]
```diff
@@ -242,6 +242,25 @@ paths:
...
+    post:
+      summary: Exchange authorization code from issuer oauth provider
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExchangeAuthorizationCodeResponse'
+      operationId: exchange-authorization-code-request
```
Comment (aholovko) [inline docs/v1/openapi.yaml:255]: change "exchange-authorization-code-request" to "exchange-authorization-code" for generated handler to be `ExchangeAuthorizationCode`

File: pkg/service/oidc4vc/oidc4vc_service_exchnage_code_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		mux.HandleFunc("/token", func(writer http.ResponseWriter, request *http.Request) {
+			writer.Header().Set("content-type", "application/json")
+			writer.WriteHeader(200)
+			_, _ = writer.Write([]byte("{   \"access_token\": \"SlAV32hkKG\",   \"token_type\": \"Bearer\",   \"refresh_token\": \"8xLOxBtZp8\",   \"expires_in\": 3600,   \"id_token\": \"1234\"  }")) // nolint
+		})
+		httpSrv.Handler = mux
+		_ = httpSrv.ListenAndServe()
+	}()
+
+	time.Sleep(2 * time.Second)
```
Comment (aholovko) [inline pkg/service/oidc4vc/oidc4vc_service_exchnage_code_test.go:135]: this doesn't look like a unit test
try to introduce abstraction for `oauthConfig` with its `Exchange` method and use mock

**Review by aholovko — CHANGES_REQUESTED** [review]
Additional changes:
* fix "exchnage" in `pkg/service/oidc4vc/oidc4vc_service_exchnage_code_test.go`
* update `controller_e2e_flows_test.go` "integration" test with additional step for token endpoint

**Review by aholovko — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**